src/net_processing.cpp

```diff
@@ -1888,16 +1888,23 @@ void static ProcessOrphanTx(CConnman* connman, std::set<uint256>& orphan_work_se
 }
 
 /**
- *  A block has been processed. If this is the first time we've seen the block,
+ *  A block has been processed. If the block was failed anti-dos / mutation checks, then
+ *  call BlockChecked() to maybe punish peer. If this is the first time we've seen the block,
  *  update the node's nLastBlockTime. Otherwise erase it from mapBlockSource.
```

Comment:
The documentation strikes me as repetitive with the code. Perhaps it could be phrased differently such that if the code changes the comment doesn't need to be updated.

nit: s/was failed/failed